include/seqan3/alignment/pairwise_alignment_edit_distance_unbanded.hpp

```diff
@@ -0,0 +1,551 @@
+    static constexpr bool use_max_errors = detail::max_errors_concept<align_config_t>;
+    //!\brief Whether the alignment is a global alignment or not.
+    static constexpr bool is_global = detail::global_config_concept<align_config_t>;
+    //!\brief Whether the alignment is a semi-global alignment or not.
+    static constexpr bool is_semi_global = detail::semi_global_config_concept<align_config_t>;
+
+    //!\brief How to pre-initialize hp.
+    static constexpr word_type hp0 = is_global ? 1 : 0;
+
+    static_assert(8*sizeof(word_type) <= 64, "we assume at most uint64_t as word_type");
+    static_assert((is_global && !is_semi_global) || (!is_global && is_semi_global), "Either set global or semi-global");
+
+    //!\brief The score of the alignment.
+    score_type _score{};
+    //!\brief The mask with a bit set at the position where the score change.
+    //!\details If #use_max_errors is true this corresponds to the last active cell.
+    word_type score_mask{0};
+    //!\brief The machine words which stores the positive vertical differences.
+    std::vector<word_type> vp{};
+    //!\brief The machine words which stores the negative vertical differences.
+    std::vector<word_type> vn{};
+    //!\brief The machine words which translate a letter of the query into a bit mask.
+    //!\details Each bit position which is true (=1) corresponds to a match of a letter in the query at this position.
+    std::vector<word_type> bit_masks{};
+
+    /*!\name Only used when use_max_errors is true
+     * \{
+     */
+    //!\brief Which score value is considered as a hit?
+    unsigned max_errors{255};
+    //!\brief The block containing the last active cell.
+    unsigned last_block{0};
+    //!\brief A mask with a bit set on the position of the last row.
+    word_type last_score_mask{};
+    //!\}
+
+    //!\brief The current position in the database.
+    database_iterator database_it{};
+    //!\brief The end position of the database.
+    database_iterator database_it_end{};
+
+    //!\brief The state of one column computation.
+    struct state_type
+    {
+        //!\copydoc pairwise_alignment_edit_distance_unbanded::vp
+        std::vector<word_type> vp{};
+        //!\copydoc pairwise_alignment_edit_distance_unbanded::vn
+        std::vector<word_type> vn{};
+    };
+    //!\brief The collection of each computation step.
+    std::vector<state_type> states{};
+    //!\brief Add a computation step
+    void add_state()
+    {
+        states.push_back(state_type{vp, vn});
+    }
+
+public:
+
+    /*!\name Constructors
+     * \{
+     */
+    /*!\brief Constructor
+     * \param[in] _database \copydoc database
+     * \param[in] _query    \copydoc query
+     * \param[in] _config   \copydoc config
+     */
+    pairwise_alignment_edit_distance_unbanded(
+        database_t _database,
+        query_t _query,
+        align_config_t _config)
+        : database{std::forward<database_t>(_database)},
```

Comment:
forward is superfluous. Either we copy a view or a l-value ref for the sequences.

Reply:
I also want to bind containers.

Reply:
I see the point now.

Reply:
The comment was not addressed. the forward has no effect, because there is no forwarding reference. Also the indention is weird, maybe this would be more legible:
```cpp
    pairwise_alignment_edit_distance_unbanded(database_t _database, query_t _query, align_config_t _config) :
        database{std::forward<database_t>(_database)},
        query{std::forward<query_t>(_query)},
```

Reply:
The forward is correct, because the type deduction:

https://github.com/seqan/seqan3/blob/ae2fad8773728cfc567fff15d7c0dde15a663a46/include/seqan3/alignment/pairwise_alignment_edit_distance_unbanded.hpp#L464-L466